### ISSUE TITLE
Widget fails over 4G when Data Saver enabled

### ISSUE THREAD

**Issue by tathamoddie** [home-assistant/android#434]
**Home Assistant Android version:** 1.7.2-128

**Android version:** 10, with 5 Feb 2020 security patches

**Phone model:** Google Pixel 3a

**Home Assistant version:** 0.105.4

**Last working Home Assistant release (if known):** 0.104

**Description of problem:**
I have added a widget that fires `switch.turn_on`. It works as expected on local Wi-Fi. When I turn off Wi-Fi, the widget fails (does the spinner for ~1sec, then goes red).

The full app view still works fine off Wi-Fi (via Home Assistant Cloud).

This was working in the previous version of the Android app, before the widget's visual re-design.

I have removed and recreated the widget multiple times.

**Traceback (if applicable):**

I don't know how to extract logs from the Android app for the widget call.

```

```

**Screenshot of problem:**

**Additional information:**

**Comment by dshokouhi** [comment]
Are you getting any errors in the HA log when you click on the widget?  How do you have the app connected to HA in App Configuration?  I assume you are using Nabu Casa remote here?

**Comment by tathamoddie** [comment]
I've narrowed it down to the "Data Saver" function.

Phone on local Wi-Fi: app and widget both work ✅

Phone on 4G, data saver off: app and widget both work ✅

Phone on 4G, data saver on: app works, widget fails ❌

Remote is via NabuCasa cloud, yes.

Why did I think it worked in a previous version? I don't use Data Saver in-country. I got the 1.7 app update while travelling, noticed the new widget design, and tried to test it. Failed. Left it until I got home. Still failed. Just noticed this morning that I still had Data Saver enabled.
